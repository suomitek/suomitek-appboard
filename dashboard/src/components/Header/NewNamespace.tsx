import * as React from "react";
import Modal from "react-modal";
import { ErrorSelector } from "../ErrorAlert";

interface INewNamespaceProps {
  modalIsOpen: boolean;
  namespace: string;
  onConfirm: () => any;
  closeModal: () => any;
  onChange: (e: React.FormEvent<HTMLInputElement>) => void;
  error?: Error;
}

interface INewNamespaceState {
  modalIsOpen: boolean;
  namespace: string;
}

class NewNamespace extends React.Component<INewNamespaceProps, INewNamespaceState> {
  public state: INewNamespaceState = {
    modalIsOpen: this.props.modalIsOpen,
    namespace: "",
  };

  public render() {
    return (
      <Modal
        className="centered-modal"
        isOpen={this.props.modalIsOpen}
        onRequestClose={this.props.closeModal}
        contentLabel="Modal"
      >
        {this.props.error && (
          <ErrorSelector
            error={this.props.error}
            resource={`namespace ${this.state.namespace}`}
            action="create"
          />
        )}

        <div>
          <div>
            <label htmlFor="new-ns">
              <div className="row">
                <div className="col-3 block">
                  <div className="centered">Namespace</div>
                </div>
                <div className="col-9 margin-t-big margin-l-big">
                  <input
                    id="new-ns"
                    onChange={this.props.onChange}
                    value={this.props.namespace}
                    type="text"
                  />
                  <span className="description">Introduce the name of the new namespace</span>
                </div>
              </div>
            </label>
          </div>
          <div className="margin-t-normal button-row">
            <button className="button" onClick={this.props.closeModal}>
              Cancel
            </button>
            <button className="button button-primary" type="submit" onClick={this.onConfirm}>
              Create
            </button>
          </div>
        </div>
      </Modal>
    );
  }

  private onConfirm = () => {
    this.setState({ namespace: this.props.namespace });
    this.props.onConfirm();
  };
}

export default NewNamespace;
