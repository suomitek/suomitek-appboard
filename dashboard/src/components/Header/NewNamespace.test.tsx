import { shallow } from "enzyme";
import * as React from "react";
import NewNamespace from "./NewNamespace";

const defaultProps = {
  modalIsOpen: true,
  namespace: "foo",
  onConfirm: jest.fn(),
  closeModal: jest.fn(),
  onChange: jest.fn(),
  error: new Error("boom!"),
};

it("renders the snapshot", () => {
  const wrapper = shallow(<NewNamespace {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
