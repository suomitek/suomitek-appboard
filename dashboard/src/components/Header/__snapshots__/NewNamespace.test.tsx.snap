// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="centered-modal"
  closeTimeoutMS={0}
  contentLabel="Modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ErrorSelector
    action="create"
    error={[Error: boom!]}
    resource="namespace "
  />
  <div>
    <div>
      <label
        htmlFor="new-ns"
      >
        <div
          className="row"
        >
          <div
            className="col-3 block"
          >
            <div
              className="centered"
            >
              Namespace
            </div>
          </div>
          <div
            className="col-9 margin-t-big margin-l-big"
          >
            <input
              id="new-ns"
              onChange={[MockFunction]}
              type="text"
              value="foo"
            />
            <span
              className="description"
            >
              Introduce the name of the new namespace
            </span>
          </div>
        </div>
      </label>
    </div>
    <div
      className="margin-t-normal button-row"
    >
      <button
        className="button"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button button-primary"
        onClick={[Function]}
        type="submit"
      >
        Create
      </button>
    </div>
  </div>
</Modal>
`;
